### PR TITLE
S port scale fix

### PR DESCRIPTION
**Pull request by fredmcc**
Fixed some issues on FrSky S.Port telemetry

1. Fixed scaling on GPS coords and Altitude
2. Fixed heading data
3. Correctly formatted GPS date/time information 

Tested on a Pixracer with Taranis+

### CODE REVIEW

File: src/drivers/frsky_telemetry/sPort_data.c [hunk 1]
```diff
@@ -302,13 +317,31 @@ void sPort_send_GPS_CRS(int uart)
 	/* send course */
 
 	/* convert to 30 bit signed magnitude degrees*6E5 with MSb = 1 and bit 30=sign */
-	uint32_t iYaw = 100 * global_pos->yaw;
+	int32_t iYaw = local_pos->yaw * 18000.0f / M_PI_F;
```
Comment (LorenzMeier) [inline src/drivers/frsky_telemetry/sPort_data.c]: Why is local position necessary here?

Comment (fredmcc) [inline src/drivers/frsky_telemetry/sPort_data.c]: It's a good question that maybe someone can help me with. For telemetry we are really looking for instantaneous data like in the VFR HUD mavlink message. I found during testing that global position data seems to not be provided for some time after start up and therefore gives no indications, even though the mavlink messages do.  local position seems to provide the data more quickly for heading. I haven't been able to really work out the difference between the different sources. If someone can point me in the right direction to find the appropriate message source I would appreciate it.

Comment (fredmcc) [inline src/drivers/frsky_telemetry/sPort_data.c]: @LorenzMeier Should I return this back to using global position so we can merge the rest? I definately think we should fix the GPS and Altitude errors

**Comment by dagar** [comment]
@fredmcc can you rebase? I think that should clean up the build failure.

**Comment by fredmcc** [comment]
@dagar rebased and now builds ok.

**Comment by LorenzMeier** [comment]
Sorry for the latency on this - it fell through the cracks. I'm merging now as is, but there is room for optimization. Are you interested in polishing the S.Port output further? If yes, I would like to encourage you to participate on the dev calls so we can pair you with someone with more system context:

https://dev.px4.io/en/contribute/#weekly-dev-call

**Pull request merged**